src/strategies.jl

```
@@ -124,55 +124,89 @@ end
 
 # ---------------------------------------------------------------------------------
 
+abstract StateUpdater
+
+immutable NoUpdater <: StateUpdater end
+state!(model, su::NoUpdater, obs...) = return
+
+immutable BackpropUpdater <: StateUpdater end
+function state!(model, su, target, input)
```

Should this be `state!(model, su::BackpropUpdater, target, input)`? Otherwise we could get silent errors here if some garbage is passed in as `su`

---

yes... though doubtful that it would ever cause an actual problem.  I'll change it.